packages/server/src/index.ts

```diff
@@ -15,6 +15,23 @@ export interface PheroServiceDefinition {
 
 export type PheroContext<T = {}> = T
 
+/**
+ * By default all middleware input is parsed by Phero. With PheroUnchecked<T> you can
+ * mark a property within your middleware context, so that Phero will skip it during
+ * the parsing step. E.g.
+ * @example
+ * ```typescript
+ *   async function getArticle(ctx: PheroContext<{ db: PheroUnchecked<RealDB> }>): Promise<string> {
+ *     return ctx.db.query()
+ *   }
+ *
+ *   async function myMiddleware(context: PheroContext, next: PheroNextFunction<{ db: PheroUnchecked<DBClient> }>) {
+ *     await next({ db: new DBClient() })
+ *   }
+ * ```
+ */
```

Comment:
"so that Phero will skip it during the parsing step" sounds like Phero will just omit the type. I also don't think users will understand what the "parsing step" refers to.

The example also has issues. JSDoc isn't markdown, so the ` ```typescript` prefix should be omitted. Also the example shows incorrect usage, since we are providing `DBClient` but asking for `RealDB`.

How about this?

```ts
/**
 * PheroUnchecked<T> is a special type that tells Phero it should not check at
 * runtime if the types actually match what we expect. It may only be used
 * within {@link PheroContext}, and must be provided by a middleware. This is
 * useful for putting types in {@link PheroContext} that would normally be
 * unsupported, such as a classes.
 *
 * @example
 * async function getArticle(ctx: PheroContext<{ db: PheroUnchecked<DBClient> }>): Promise<string> {
 *   return ctx.db.query()
 * }
 *
 * async function myMiddleware(context: PheroContext, next: PheroNextFunction<{ db: PheroUnchecked<DBClient> }>) {
 *   await next({ db: new DBClient() })
 * }
 */
```

Reply:
- "at runtime"
☝️ in case Phero will not complain if this does compile:

``` 
*   async function getArticle(ctx: PheroContext<{ db: PheroUnchecked<RealDB> }>): Promise<string> {
 *     return ctx.db.query()
 *   }
 *
 *   async function myMiddleware(context: PheroContext, next: PheroNextFunction<{ db: PheroUnchecked<DBClient> }>) {
 *     await next({ db: new DBClient() })
 *   }
 ```
 
 Notice the RealDB/DBClient typo